src/sentry/api/serializers/models/release.py

```diff
@@ -311,16 +322,44 @@ def get_attrs(self, item_list, user, **kwargs):
         if with_health_data and no_snuba:
             raise TypeError("health data requires snuba")
 
+        self.has_adoption_stage = features.has(
+            "organizations:release-adoption-stage", item_list[0].organization_id
+        )
+        if self.has_adoption_stage:
+            release_project_envs = (
+                ReleaseProjectEnvironment.objects.filter(release__in=item_list)
+                .select_related("release")
```

Comment:
We need to select `project` here too, since it's used in `_get_release_adoption_stage`

Reply:
There is a project filter a couple lines below this that should do the trick I think?

Reply:
I don't know if that ends up doing the same thing as `select_related`. When you call `release_project_env.project` later on I think it'll make another query.

Reply:
Oh, sorry, I interpreted your first message as adding filter by project here. Good call about the `select_related`, I'll add that in.